Assets/HoloToolkit/SpatialMapping/Scripts/TapToPlaceOnSpatialMapping.cs

```diff
@@ -0,0 +1,255 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System.Collections.Generic;
+using UnityEngine;
+using HoloToolkit.Unity;
+using HoloToolkit.Unity.InputModule;
+
+namespace HoloToolkit.Unity.SpatialMapping
+{
+    /// <summary>
+    /// The TapToPlaceOnSpatialMapping class is a basic way to enable users to move objects 
+    /// and place them on real world surfaces.
+    /// Put this script on the object you want to be able to move. 
+    /// Users will be able to tap objects, gaze elsewhere, and perform the tap gesture again to place.
+    /// This script is used in conjunction with GazeManager, WorldAnchorManager, and SpatialMappingManager.
+    /// </summary>
+    [RequireComponent(typeof(Collider))]
+    [RequireComponent(typeof(Interpolator))]
+    public class TapToPlaceOnSpatialMapping : MonoBehaviour, IInputClickHandler
```

Comment:
I think we inherit from the `TapToPlace` so we don't have so much duplicated code.

Reply:
The point is precisely not to :) We are separating as much cross feature as we can :)

Reply:
This script requires `IInputClickHandler`, which is in the Input folder as well. Inheriting from `TapToPlace` wouldn't add any additional dependencies.

Reply:
Yeah, we're already using the `HoloToolkit.Unity.InputModule` so it kinda defeats the purpose.

Reply:
I will be removing these classes as I am not convinced this is the right way anyway to handle this.